crates/guest-rust/rt/src/async_support.rs

```diff
@@ -107,7 +111,11 @@ unsafe fn poll(state: *mut FutureState) -> Poll<()> {
                         (*state).tasks = None;
                         break Poll::Ready(());
                     }
-                    Poll::Pending => break Poll::Pending,
+                    Poll::Pending => {
+                        if !waker.0.load(Ordering::Relaxed) {
+                            break Poll::Pending;
+                        }
```

Comment:
I forget how all the bits line up here, but this might be a case where it'd be good to yield to the host? E.g. the async export would return a code indicating that it's yielding. (although I know `yield` is also an imported function, so now I'm not sure if there's a return from the top-level function that means yielding is happening... if that's not the case then this looks reasonable to me)

Reply:
Yeah, we could definitely yield here if desired, although `wasip3-prototyping` doesn't support guests returning `STATUS_YIELDED` yet due to it being a relatively newer addition to the spec I haven't gotten around to yet.  Shouldn't be hard to support, though.

Reply:
Makes sense, mind leaving a comment indicating that we probably want to transition to that?